### ISSUE TITLE
Parameters not validating interfaces correctly.

### ISSUE THREAD

**Issue by wdamien** [Microsoft/TypeScript#9796]
**TypeScript Version:**  1.8.10

**Code**

```ts
interface IFoo {
	bar();
}

class Foo implements IFoo {
	public bar() { }
}

class Baz {
	public bar() { return "Don't eat me."; }
}

class EatIFoo {
	
	public Eat(foo:IFoo) {
		console.log(foo.bar());
	}
}

var eat = new EatIFoo();
eat.Eat(new Baz()); // Should cause an error. since Baz is not an IFoo
eat.Eat(new Foo()); // Works as expected!
```

**Expected behavior:**
- The compiler should not allow an instance of `new Baz()` to be passed into `eat.Eat();`

**Actual behavior:** 
- The compiler lets `new Baz()` in as a parameter even though its not an instance of `IFoo`.

**Comment by sandersn** [comment]
See [this FAQ answer](https://github.com/Microsoft/TypeScript/wiki/FAQ#what-is-structural-typing). `Baz` is *structurally* an `IFoo` because it has a `bar` method that is structurally compatible with `IFoo.bar`.
